### ISSUE TITLE
Error Handler for mock does not exist

### ISSUE THREAD

**Issue by illuxio** [gstroup/apimocker#29]
apimocker throws ```Error: ENOENT: no such file or directory``` if no mock file exists.

It would be nice to have an callback to handle this.

**Comment by gstroup** [comment]
Hi, that is an interesting idea.  Can you explain more what you were thinking?  What would you like to do in that case?
I don't think we want to pass in the callback in the config.json file.  I supposed we could expose a callback property that you could set from your code.  Is that what you're looking for?

**Comment by ztsmith** [comment]
A more general solution could be to expose the express middleware. So you could for instance add a 404 middleware handler into the express pipeline.

**Comment by illuxio** [comment]
Exposing express would be nice or have an access option. 

My personal problem is currently the inconsistent error handling as shown in the code snippet.

```
    if (options.contentType) {
      res.header('Content-Type', options.contentType);
      fs.readFile(mockPath, {encoding: "utf8"}, function(err, data) {
        if (err) { throw err; } // Throws error if mock file is missing
        var buff = new Buffer(data, 'utf8');
        res.status(options.httpStatus).send(buff);
      });
    } else {
      res.status(options.httpStatus).sendfile(encodeURIComponent(options.mockFile), {root: apiMocker.options.mockDirectory});  // Throws no error if mock file is missing
    }
```

**Comment by gstroup** [comment]
Fixed in v0.4.16.  Will return a 404 by default, or you can specify another httpStatus to return.
